### ISSUE TITLE
plotting w/ qt not working on osx

### ISSUE THREAD

**Issue by beckermr** [conda-forge/matplotlib-feedstock#236]
Issue:

I am getting this in a notebook

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-9-77e93c4f964d> in <module>
----> 1 plt.savefig("m-e2e-007-grid.png")

~/miniconda3/envs/anl/lib/python3.7/site-packages/matplotlib/pyplot.py in savefig(*args, **kwargs)
    721 def savefig(*args, **kwargs):
    722     fig = gcf()
--> 723     res = fig.savefig(*args, **kwargs)
    724     fig.canvas.draw_idle()   # need this if 'transparent=True' to reset colors
    725     return res

~/miniconda3/envs/anl/lib/python3.7/site-packages/matplotlib/figure.py in savefig(self, fname, transparent, **kwargs)
   2201             self.patch.set_visible(frameon)
   2202 
-> 2203         self.canvas.print_figure(fname, **kwargs)
   2204 
   2205         if frameon:

~/miniconda3/envs/anl/lib/python3.7/site-packages/matplotlib/backend_bases.py in print_figure(self, filename, dpi, facecolor, edgecolor, orientation, format, bbox_inches, **kwargs)
   2091                     orientation=orientation,
   2092                     bbox_inches_restore=_bbox_inches_restore,
-> 2093                     **kwargs)
   2094             finally:
   2095                 if bbox_inches and restore_bbox:

~/miniconda3/envs/anl/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py in print_png(self, filename_or_obj, metadata, pil_kwargs, *args, **kwargs)
    512         }
    513 
--> 514         FigureCanvasAgg.draw(self)
    515         if pil_kwargs is not None:
    516             from PIL import Image

~/miniconda3/envs/anl/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py in draw(self)
    390         with RendererAgg.lock, \
    391              (self.toolbar._wait_cursor_for_draw_cm() if self.toolbar
--> 392               else nullcontext()):
    393             self.figure.draw(self.renderer)
    394             # A GUI class may be need to update a window using this draw, so

~/miniconda3/envs/anl/lib/python3.7/contextlib.py in __enter__(self)
    110         del self.args, self.kwds, self.func
    111         try:
--> 112             return next(self.gen)
    113         except StopIteration:
    114             raise RuntimeError("generator didn't yield") from None

~/miniconda3/envs/anl/lib/python3.7/site-packages/matplotlib/backend_bases.py in _wait_cursor_for_draw_cm(self)
   2771         if self._draw_time - last_draw_time > 1:
   2772             try:
-> 2773                 self.set_cursor(cursors.WAIT)
   2774                 yield
   2775             finally:

~/miniconda3/envs/anl/lib/python3.7/site-packages/matplotlib/backends/backend_webagg_core.py in set_cursor(self, cursor)
    379     def set_cursor(self, cursor):
    380         if cursor != self.cursor:
--> 381             self.canvas.send_event("cursor", cursor=cursor)
    382         self.cursor = cursor
    383 

~/miniconda3/envs/anl/lib/python3.7/site-packages/matplotlib/backends/backend_webagg_core.py in send_event(self, event_type, **kwargs)
    344 
    345     def send_event(self, event_type, **kwargs):
--> 346         self.manager._send_event(event_type, **kwargs)
    347 
    348 

AttributeError: 'NoneType' object has no attribute '_send_event'

```

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /Users/beckermr/miniconda3/envs/anl:
#
# Name                    Version                   Build  Channel
absl-py                   0.9.0                    py37_0    conda-forge
alabaster                 0.7.12                     py_0    conda-forge
appdirs                   1.4.3                      py_1    conda-forge
appnope                   0.1.0                 py37_1000    conda-forge
astropy                   4.0              py37h0b31af3_1    conda-forge
attrs                     19.3.0                     py_0    conda-forge
babel                     2.8.0                      py_0    conda-forge
backcall                  0.1.0                      py_0    conda-forge
beautifulsoup4            4.8.2                    py37_0    conda-forge
black                     19.10b0                  py37_0    conda-forge
bleach                    3.1.1                      py_0    conda-forge
bzip2                     1.0.8                h0b31af3_2    conda-forge
ca-certificates           2019.11.28           hecc5488_0    conda-forge
camb                      1.1.1            py37h1ddc27c_0    conda-forge
cctools                   927.0.2              h5ba7a2e_4    conda-forge
certifi                   2019.11.28       py37hc8dfbb8_1    conda-forge
cffi                      1.14.0           py37h356ff06_0    conda-forge
cfgv                      3.0.0                      py_0    conda-forge
chardet                   3.0.4                 py37_1003    conda-forge
clang                     9.0.1           default_hf57f61e_0    conda-forge
click                     7.0                        py_0    conda-forge
corrfunc                  2.3.3            py37h4d8ce68_1    conda-forge
cryptography              2.8              py37hafa8578_1    conda-forge
cycler                    0.10.0                     py_2    conda-forge
dbus                      1.13.6               h2f22bb5_0    conda-forge
decorator                 4.4.2                      py_0    conda-forge
defusedxml                0.6.0                      py_0    conda-forge
differentiable-smhm       0.0.dev0                  dev_0    <develop>
docutils                  0.16                     py37_0    conda-forge
editdistance              0.5.3            py37h4a8c4bd_0    conda-forge
eigen                     3.3.7             ha1b3eb9_1001    conda-forge
emcee                     3.0.2              pyh5ca1d4c_0    conda-forge
entrypoints               0.3                   py37_1000    conda-forge
expat                     2.2.9                h4a8c4bd_2    conda-forge
fastcache                 1.1.0            py37h0b31af3_0    conda-forge
fftw                      3.3.8           nompi_h44ae4c8_1110    conda-forge
firecrown                 0.1.0                     dev_0    <develop>
fitsio                    1.1.1            py37h93e8f8d_0    conda-forge
flake8                    3.7.9                    py37_0    conda-forge
freetype                  2.10.0               h24853df_1    conda-forge
future                    0.18.2                   py37_0    conda-forge
galsim                    2.2.3            py37hc0fbbca_0    conda-forge
gettext                   0.19.8.1          h46ab8bc_1002    conda-forge
gfortran_impl_osx-64      7.3.0                hf4212f2_2    conda-forge
gfortran_osx-64           7.3.0                h22b1bf0_7    conda-forge
glib                      2.58.3          py37h577aef8_1002    conda-forge
gmp                       6.2.0                h4a8c4bd_2    conda-forge
gmpy2                     2.1.0b1          py37h4160ff4_0    conda-forge
gsl                       2.6                  ha2d443c_0    conda-forge
h5py                      2.10.0          nompi_py37h106b333_102    conda-forge
halotools                 0.7              py37h4f17bb1_0    conda-forge
hdf5                      1.10.5          nompi_h3e39495_1104    conda-forge
icu                       64.2                 h6de7cb9_1    conda-forge
identify                  1.4.11                     py_0    conda-forge
idna                      2.9                        py_1    conda-forge
imagesize                 1.2.0                      py_0    conda-forge
importlib_metadata        1.5.0                    py37_0    conda-forge
ipykernel                 5.1.4            py37h5ca1d4c_0    conda-forge
ipython                   7.13.0           py37h5ca1d4c_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.5.1                      py_0    conda-forge
isl                       0.19                          0    conda-forge
jax                       0.1.59             pyh5ca1d4c_0    conda-forge
jaxlib                    0.1.39           py37h5ca1d4c_0    conda-forge
jedi                      0.16.0                   py37_0    conda-forge
jinja2                    2.11.1                     py_0    conda-forge
joblib                    0.14.1                     py_0    conda-forge
jpeg                      9c                h1de35cc_1001    conda-forge
jsonschema                3.2.0                    py37_0    conda-forge
jupyter                   1.0.0                      py_2    conda-forge
jupyter_client            6.0.0                      py_0    conda-forge
jupyter_console           6.0.0                      py_0    conda-forge
jupyter_core              4.6.3                    py37_0    conda-forge
kiwisolver                1.1.0            py37ha1b3eb9_0    conda-forge
krb5                      1.16.4               h1752a42_0    conda-forge
ld64                      450.3                h3c32e8a_4    conda-forge
libblas                   3.8.0               15_openblas    conda-forge
libcblas                  3.8.0               15_openblas    conda-forge
libclang                  9.0.1           default_hf57f61e_0    conda-forge
libcurl                   7.68.0               h709d2b2_0    conda-forge
libcxx                    9.0.1                         1    conda-forge
libedit                   3.1.20170329      hcfe32e1_1001    conda-forge
libffi                    3.2.1             h6de7cb9_1006    conda-forge
libgfortran               4.0.0                         2    conda-forge
libiconv                  1.15              h01d97ff_1005    conda-forge
liblapack                 3.8.0               15_openblas    conda-forge
libllvm8                  8.0.1                h770b8ee_0    conda-forge
libllvm9                  9.0.1                ha1b3eb9_0    conda-forge
libopenblas               0.3.8                h3d69b6c_0    conda-forge
libpng                    1.6.37               h2573ce8_0    conda-forge
libpq                     12.2                 h554dc5a_0    conda-forge
libprotobuf               3.11.4               hd174df1_0    conda-forge
libsodium                 1.0.17               h01d97ff_0    conda-forge
libssh2                   1.8.2                hcdc9a53_2    conda-forge
llvm-openmp               9.0.1                h28b9765_2    conda-forge
llvmlite                  0.31.0           py37h05045ef_0    conda-forge
lsstdesc.coord            1.2.0            py37ha1b3eb9_0    conda-forge
markupsafe                1.1.1            py37h0b31af3_0    conda-forge
matplotlib                3.2.0                         1    conda-forge
matplotlib-base           3.2.0            py37h11da6c2_1    conda-forge
mccabe                    0.6.1                      py_1    conda-forge
meds                      0.9.10           py37h3b54f70_0    conda-forge
mistune                   0.8.4           py37h0b31af3_1000    conda-forge
mock                      3.0.5                    py37_0    conda-forge
more-itertools            8.2.0                      py_0    conda-forge
mpc                       1.1.0             h4160ff4_1006    conda-forge
mpfr                      4.0.2                h44b798e_0    conda-forge
mpi                       1.0                     openmpi    conda-forge
mpi4py                    3.0.3            py37h8b5c983_0    conda-forge
mpmath                    1.1.0                      py_0    conda-forge
mypy_extensions           0.4.3                    py37_0    conda-forge
nbconvert                 5.6.1                    py37_0    conda-forge
nbformat                  5.0.4                      py_0    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
ngmix                     1.3.7                    py37_0    conda-forge
nodeenv                   1.3.5                      py_0    conda-forge
notebook                  6.0.3                    py37_0    conda-forge
nspr                      4.20              h0a44026_1000    conda-forge
nss                       3.47                 hc0980d9_0    conda-forge
numba                     0.48.0           py37h4f17bb1_0    conda-forge
numpy                     1.18.1           py37hde6bac1_0    conda-forge
numpydoc                  0.9.2                      py_0    conda-forge
openmpi                   4.0.2                hed52333_4    conda-forge
openssl                   1.1.1d               h0b31af3_0    conda-forge
opt_einsum                3.2.0                      py_0    conda-forge
packaging                 20.1                       py_0    conda-forge
pandas                    1.0.1            py37h4f17bb1_0    conda-forge
pandoc                    2.9.2                         0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
parso                     0.6.2                      py_0    conda-forge
pathspec                  0.7.0                      py_0    conda-forge
patsy                     0.5.1                      py_0    conda-forge
pcre                      8.44                 h4a8c4bd_0    conda-forge
pexpect                   4.8.0                    py37_0    conda-forge
pickleshare               0.7.5                 py37_1000    conda-forge
pip                       20.0.2                     py_2    conda-forge
pluggy                    0.13.0                   py37_0    conda-forge
pre-commit                2.1.1                    py37_0    conda-forge
prometheus_client         0.7.1                      py_0    conda-forge
prompt-toolkit            3.0.4                      py_0    conda-forge
prompt_toolkit            2.0.10                     py_0    conda-forge
protobuf                  3.11.4           py37h4a8c4bd_0    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
py                        1.8.1                      py_0    conda-forge
pybind11                  2.4.3            py37ha1b3eb9_1    conda-forge
pyccl                     2.0.1            py37h5a7dc0a_4    conda-forge
pycodestyle               2.5.0                      py_0    conda-forge
pycparser                 2.20                       py_0    conda-forge
pyflakes                  2.1.1                      py_0    conda-forge
pygments                  2.6.1                      py_0    conda-forge
pyopenssl                 19.1.0                     py_1    conda-forge
pyparsing                 2.4.6                      py_0    conda-forge
pyqt                      5.12.3           py37he22c54c_1    conda-forge
pyqt5-sip                 4.19.18                  pypi_0    pypi
pyqtwebengine             5.12.1                   pypi_0    pypi
pyrsistent                0.15.7           py37h0b31af3_0    conda-forge
pysocks                   1.7.1                    py37_0    conda-forge
pytest                    5.3.5                    py37_1    conda-forge
pytest-mpi                0.3                           0    conda-forge
python                    3.7.6           hfa4aa89_4_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.7                     1_cp37m    conda-forge
pytz                      2019.3                     py_0    conda-forge
pyyaml                    5.3              py37h0b31af3_0    conda-forge
pyzmq                     19.0.0           py37h4bf09a9_0    conda-forge
qt                        5.12.5               h514805e_3    conda-forge
qtconsole                 4.7.1                      py_0    conda-forge
qtpy                      1.9.0                      py_0    conda-forge
readline                  8.0                  hcfe32e1_0    conda-forge
regex                     2020.2.20        py37h0b31af3_0    conda-forge
requests                  2.23.0                   py37_0    conda-forge
ruamel.yaml               0.16.6           py37h0b31af3_0    conda-forge
ruamel.yaml.clib          0.2.0            py37h0b31af3_0    conda-forge
ruamel.yaml.jinja2        0.2.4                      py_1    conda-forge
sacc                      0.2.3                      py_0    conda-forge
schwimmbad                0.3.0                    py37_2    conda-forge
scikit-learn              0.22.2.post1     py37h3dc85bc_0    conda-forge
scipy                     1.4.1            py37h82752d6_0    conda-forge
seaborn                   0.10.0                     py_1    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                46.0.0           py37hc8dfbb8_1    conda-forge
six                       1.14.0                     py_1    conda-forge
snowballstemmer           2.0.0                      py_0    conda-forge
soupsieve                 1.9.4                    py37_0    conda-forge
sphinx                    2.4.4                      py_0    conda-forge
sphinxcontrib-applehelp   1.0.2                      py_0    conda-forge
sphinxcontrib-devhelp     1.0.2                      py_0    conda-forge
sphinxcontrib-htmlhelp    1.0.3                      py_0    conda-forge
sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
sphinxcontrib-qthelp      1.0.3                      py_0    conda-forge
sphinxcontrib-serializinghtml 1.1.4                      py_0    conda-forge
sqlite                    3.30.1               h93121df_0    conda-forge
statsmodels               0.11.1           py37h9bfed18_1    conda-forge
sturdy-garbanzo           0.0.1                     dev_0    <develop>
sympy                     1.5.1                    py37_1    conda-forge
tapi                      1000.10.8            ha1b3eb9_4    conda-forge
terminado                 0.8.3                    py37_0    conda-forge
testpath                  0.4.4                      py_0    conda-forge
thechopper                0.1                        py_0    conda-forge
tk                        8.6.10               hbbe82c9_0    conda-forge
toml                      0.10.0                     py_0    conda-forge
tornado                   6.0.4            py37h9bfed18_1    conda-forge
tqdm                      4.43.0                     py_0    conda-forge
traitlets                 4.3.3                    py37_0    conda-forge
treecorr                  4.0.10           py37h4a8c4bd_0    conda-forge
typed-ast                 1.4.1            py37h0b31af3_0    conda-forge
typing_extensions         3.7.4.1                  py37_0    conda-forge
urllib3                   1.25.7                   py37_0    conda-forge
virtualenv                16.7.5                     py_0    conda-forge
wcwidth                   0.1.8                      py_0    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.34.2                     py_1    conda-forge
widgetsnbextension        3.5.1                    py37_0    conda-forge
wurlitzer                 2.0.0                    py37_0    conda-forge
xz                        5.2.4             h1de35cc_1001    conda-forge
yaml                      0.2.2                h0b31af3_1    conda-forge
zeromq                    4.3.2                h6de7cb9_2    conda-forge
zipp                      3.1.0                      py_0    conda-forge
zlib                      1.2.11            h0b31af3_1006    conda-forge
```
</details>

Note the pypi packages. I removed those, but they appear to be installed anyways. This seems like a qt issue maybe?

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

     active environment : anl
    active env location : /Users/beckermr/miniconda3/envs/anl
            shell level : 2
       user config file : /Users/beckermr/.condarc
 populated config files : /Users/beckermr/.condarc
          conda version : 4.8.2
    conda-build version : 3.18.12
         python version : 3.7.6.final.0
       virtual packages : __osx=10.14.6
       base environment : /Users/beckermr/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/beckermr/miniconda3/pkgs
                          /Users/beckermr/.conda/pkgs
       envs directories : /Users/beckermr/miniconda3/envs
                          /Users/beckermr/.conda/envs
               platform : osx-64
             user-agent : conda/4.8.2 requests/2.23.0 CPython/3.7.6 Darwin/18.7.0 OSX/10.14.6
                UID:GID : 501:20
             netrc file : /Users/beckermr/.netrc
           offline mode : False
```
</details>

@conda-forge/matplotlib 
@conda-forge/qt-py 
@conda-forge/pyqt 
@conda-forge/qt

**Comment by chrisburr** [comment]
Looks like: https://github.com/matplotlib/ipympl/issues/115

Does adding a sleep after the call to `savefig` workaround the issue? (Just to check if it is a race)

**Comment by beckermr** [comment]
Nope doesn't help.

**Comment by tacaswell** [comment]
I think this is fixed in 3.2.1

**Comment by tacaswell** [comment]
https://github.com/matplotlib/matplotlib/pull/16742

**Comment by dopplershift** [comment]
Closing because it sounds like this was fixed in 3.2.1, but feel free to reopen if it's still a problem.
